### DISCUSSION TITLE
Using Tabels without eloquent Models

### DISCUSSION THREAD

**Discussion by kolaente** [filamentphp/filament#2283] [Help] [answered]
I have data which comes from other sources than a database in my application. Is there a way to show that data without an eloquent model?

So far, I tried overriding `getTableRecords` and feed filament a fake eloquent collection and model but that didn't get me very far:

```php
class ExportExcel extends BaseWidget
{
    protected function getTableQuery(): Builder
    {
        return Participant::query();
    }

    public function getTableRecords(): Collection|Paginator
    {
        /** @var Participants $participantExporter */
        $participantExporter = resolve(Participants::class);

        return Collection::make(collect($participantExporter->getExportedFiles())
            ->map(function ($d) {
                return new class extends Model {
                    public function getAttributes()
                    {
                        return ['name', 'date'];
                    }
                };
            }));
    }

    protected function getTableColumns(): array
    {
        return [
            Tables\Columns\TextColumn::make('name')
                ->label('Export'),
            Tables\Columns\TextColumn::make('date')
                ->label('Erstellt'),
        ];
    }
}
```

My attempt does not feel like it should work or is an intended way to go about this, therefore I didn't try to make it work.

**Comment by ryangjchandler** [comment, marked as answer]
https://github.com/calebporzio/sushi

**Reply by michabbb** [reply]
i came to here because i was searching for "filament table without model". how can a package be the answer that requires a model ? pardon, maybe someone can enlighten me. a very simple example would be extremely helpful. thanks!

**Reply by danharrin** [reply]
You can create a model out of static data with Sushi, check the readme

Filament requires a model as we heavily rely on Eloquent and SQL to make the tables interactive

**Reply by michabbb** [reply]
okay, i was confused because the "use it" section in the readme says "Add the Sushi trait to a model." and I thought: wtf, I don´t want to use a model 🙈

**Comment by Sicklou** [comment]
So it could be use with a custom SQL query in the getRows() method, then used as an eloquent model.

If it works, it's really a good thing !

**Comment by GrishmaPatel126** [comment]
I have an application where I'm displaying a table with data fetched from an API using the Sushi package. When a user selects a record from this table, I want to open another table in a modal (or new page). This new table should fetch data from a different API, and this time, I'm not using the Sushi package. The data for the second table should be based on the values from the selected record in the initial table.

Can Somebody Help me for above query ?

**Reply by kolaente** [reply]
I would use normal eloquent relationships for that, combined with a filament relation manager.
